static/js/src/advantage/react/hooks/useLastPurchaseIds.ts

```diff
@@ -6,24 +6,31 @@ import { UserSubscriptionMarketplace } from "advantage/api/enum";
 /**
  * Get a the last purchase id by account id and period.
  */
-export const selectPurchaseIdsByMarketplaceAndPeriod = (
-  marketplace?: UserSubscription["marketplace"],
-  period?: UserSubscription["period"]
-) => (lastPurchaseIds: LastPurchaseIds) => {
-  if (
-    !marketplace ||
-    !period ||
-    marketplace === UserSubscriptionMarketplace.Free ||
-    marketplace === UserSubscriptionMarketplace.CanonicalCUBE
-  ) {
-    return null;
-  }
-  return lastPurchaseIds?.[marketplace]?.[period] ?? null;
-};
+export const selectPurchaseIdsByMarketplaceAndPeriod =
+  (
+    marketplace?: UserSubscription["marketplace"],
+    period?: UserSubscription["period"]
+  ) =>
+  (lastPurchaseIds: LastPurchaseIds) => {
+    if (
+      !marketplace ||
+      !period ||
+      marketplace === UserSubscriptionMarketplace.Free ||
+      marketplace === UserSubscriptionMarketplace.CanonicalCUBE
+    ) {
+      return null;
+    }
+    return lastPurchaseIds?.[marketplace]?.[period] ?? null;
+  };
 
 export const useLastPurchaseIds = <D = LastPurchaseIds>(
   accountId?: UserSubscription["account_id"] | null,
-  options: UseQueryOptions<LastPurchaseIds, unknown, D> = {}
+  options: UseQueryOptions<
+    LastPurchaseIds,
+    unknown,
+    D,
+    Array<string | null | undefined>
+  > = {}
```

Comment:
Hi @sowasred2012 in this case `Array<string | null | undefined>` is required because it needs to be the same type as the first param for the `useQuery` call: `["lastPurchaseIds", accountId]`.

`Array<string | null | undefined>` is fine, but is more permissive than the tuple defined so you could use `[string, string | null | undefined]` or the type it is generated from: `[string, UserSubscription["account_id"] | null | undefined]`.

This leads to some duplication with the `accountId` param given to `useLastPurchaseIds` (which is actually where the type for `UseQueryOptions` is defined).

So if you wanted to be really clever you could do something like this:

```suggestion
 export const useLastPurchaseIds = <ID extends UserSubscription["account_id"] | null | undefined, D = LastPurchaseIds>(
  accountId: ID,
  options: UseQueryOptions<
    LastPurchaseIds,
    unknown,
    D,
    [string, ID]
  > = {}
) => {
```

Hope that helps!

Reply:
Thanks Huw! I don't quite understand it, but the linter seems happy with it, and I trust you :rofl: